src/components/profile/EditAddress.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { fetchUserProfile, updateUserProfileData } from "../../redux/reducers/profileSlice";
+import ProfileMenu from "./ProfileMenu";
+import { showErrorMessage, showSuccessMessage } from "../../utils/toast";
+
+export default function UpdateAddress () {
+  const dispatch = useDispatch();
+  const { user, isLoading, isError } = useSelector((state) => state.profile);
+
+  useEffect(() => {
+    dispatch(fetchUserProfile());
+  }, [dispatch]);
+
+  const [email, setEmail] = useState("");
+  const [phoneNo, setPhoneNo] = useState("");
+  const [province, setProvince] = useState("");
+  const [district, setDistrict] = useState("");
+  const [street, setStreet] = useState("");
+
```

Comment:
This is too much, use react-hook-form instead of creating state for every input fields